src/metrics.ts

```diff
@@ -435,8 +435,14 @@ function generateExecutionVersions(terasliceStats:TerasliceStats, labels:any) {
 export function updateTerasliceMetrics(terasliceStats: TerasliceStats): void {
     metricsRegistry.resetMetrics();
 
+    let baseURLLabel = terasliceStats.baseUrl.toString()
+
+    if (process?.env?.TERASLICE_DISPLAY_URL) {
```

Comment:
There should be no env variable handling at this level.  That should all happen at the top index.tx file.  This override should be handled there then passed down then the `terasliceStats` object is instantiated.